src/Azure/Orleans.Transactions.AzureStorage/Storage/AzureTransactionLogStorage.cs

```diff
@@ -234,19 +259,29 @@ public Task<long> GetStartRecord()
                 if (queryResult.Results.Count > 0)
                 {
                     var batchOperation = new TableBatchOperation();
-
+                    var archivalBatchOperation = new TableBatchOperation();
```

Comment:
The add and remove should be performed in the same batch operation, else they won't be atomic.
If archiving is on, we need reduce the size of the batch to half, as half of the operation will be delete, and half will be adds.

Reply:
cannot do it since batch operation requires operation under the same partition key.

Reply:
Be careful with this suggestion, because unless you rework the partition key from it's current value, then these can't go in the same batch operation

Reply:
I didn't notice the partition keys didn't match, my mistake.  We do need these to be atomic so suggest they be in the same partition.

Reply:
Can you explain the purpose of archiving? I thought it was so that the partition doesn't get extremely big and it affects the hot access to it, but if it's kept in the same one, maybe I missed the point.

Reply:
the purpose of archiving is for test purpose only. It is not designed for production use. I talked with Jason that hiding that `ArchiveLog` setting to another option in a test namespace to make this less discoverable. 

@jason-bragg was insisting on atomic operation I think it is because in that way, we won't lose transaction records which were being deleted originally. ofc there's ways to put them on another partition through two batch operations, and then handling operation failure in a way to ensure records not lost, for example use a queue to retry on failure. But we are not doing that to my understanding is because time is tight. But I will let @jason-bragg speak his reasoning to you, instead of me interpreting.

Reply:
Archiving, at this point, is mainly to assist in our recovery and correctness testing.  By deleting entries from the transaction log we loose history needed for correctness checks.  Instead of deleting the entries we (if configured) archive them.  For us to have confidence in the results, archive records need be created for all deleted transaction log entries.  There are many ways of doing this, but performing the operations atomically within the same partition is simple and fast.